backend/src/test/groovy/pt/ulisboa/tecnico/socialsoftware/tutor/tournament/service/QuizTournamentGenerateTest.groovy

```diff
@@ -0,0 +1,124 @@
+        userRepository.save(user1)
+        user1.setKey(user1.getId())
+
+        def topicDto1 = new TopicDto()
+        topicDto1.setName(TOPIC_1_NAME)
+        topic1 = new Topic(externalCourse, topicDto1)
+        topicRepository.save(topic1)
+
+        def topicDto2 = new TopicDto()
+        topicDto2.setName(TOPIC_2_NAME)
+        topic2 = new Topic(externalCourse, topicDto2)
+        topicRepository.save(topic2)
+
+        topics.add(topic1.getId())
+        topics.add(topic2.getId())
+
+        def assessment = new Assessment()
+        assessment.setTitle(ASSESSMENT_1_TITLE)
+        assessment.setStatus(Assessment.Status.AVAILABLE)
+        assessment.setCourseExecution(externalCourseExecution)
+
+        def topicConjunction = new TopicConjunction()
+        topicConjunction.addTopic(topic1)
+        topicConjunction.addTopic(topic2)
+
+        assessment.addTopicConjunction(topicConjunction)
+        assessmentRepository.save(assessment)
+
+        tournamentDto = new TournamentDto()
+        tournamentDto.setStartTime(STRING_DATE_TODAY)
+        tournamentDto.setEndTime(STRING_DATE_LATER)
+        tournamentDto.setNumberOfQuestions(NUMBER_OF_QUESTIONS)
+        tournamentDto.setState(false)
+        tournamentDto = tournamentService.createTournament(user1.getId(), topics, tournamentDto)
+
+        question1 = new Question()
+        question1.setKey(1)
+        question1.setCreationDate(LOCAL_DATE_TODAY)
+        question1.setContent(QUESTION_1_CONTENT)
+        question1.setTitle(QUESTION_1_TITLE)
+        question1.setStatus(Question.Status.AVAILABLE)
+        question1.setCourse(externalCourse)
+        question1.addTopic(topic1)
+        question1.addTopic(topic2)
+        questionRepository.save(question1)
+    }
+
+    def "1 student join a tournament" () {
+        given:
+        tournamentService.joinTournament(user1.getId(), tournamentDto, "")
+
+        when:
+        tournamentService.solveQuiz(user1.getId(), tournamentDto)
+
+        then: "the correct quiz is inside the repository"
+        quizRepository.count() == 1L
+        def result = quizRepository.findAll().get(0)
+        result.getId() != null
+        result.getTitle() == ("Tournament " + tournamentDto.getId() + " Quiz")
+        result.getType() == Quiz.QuizType.TOURNAMENT
+        DateHandler.toISOString(result.getConclusionDate()) == STRING_DATE_LATER
+        result.getQuizQuestions().size() == NUMBER_OF_QUESTIONS
+    }
+
+    def "2 student join a tournament" () {
+        given:
+        def user2 = new User(USER_2_NAME, USER_2_USERNAME, USER_2_EMAIL, User.Role.STUDENT, false, false)
+        user2.addCourse(externalCourseExecution)
+        userRepository.save(user2)
+
+        and:
+        tournamentService.joinTournament(user1.getId(), tournamentDto, "")
+        tournamentService.joinTournament(user2.getId(), tournamentDto, "")
+
+        when: 'both students solve the quiz'
+        tournamentService.solveQuiz(user1.getId(), tournamentDto)
```

Comment:
make no sense to have tow invocations and this test does not add to the previous one

Reply:
Done